python/plugins/processing/algs/qgis/Eliminate.py

```diff
@@ -108,8 +109,8 @@ def processAlgorithm(self, progress):
             comparison = self.comparisons[self.getParameterValue(self.COMPARISON)]
             comparisonvalue = self.getParameterValue(self.COMPARISONVALUE)
 
-            selectindex = inLayer.dataProvider().fieldNameIndex(attribute)
-            selectType = inLayer.dataProvider().fields()[selectindex].type()
+            selectindex = vector.resolveFieldIndex(processLayer, attribute)
+            selectType = processLayer.dataProvider().fields()[selectindex].type()
```

Comment:
In general it's better not to use the dataProvider(). `processLayer.dataProvider().fields()` -> `processLayer.fields()`